diff --git a/src/google/protobuf/compiler/BUILD.bazel b/src/google/protobuf/compiler/BUILD.bazel
index 8cc1186b0..c417dc962 100644
--- a/src/google/protobuf/compiler/BUILD.bazel
+++ b/src/google/protobuf/compiler/BUILD.bazel
@@ -97,6 +97,7 @@ cc_library(
         "main.cc",
     ],
     copts = COPTS,
+    alwayslink = True,
     deps = [
         ":code_generator",
         ":command_line_interface",
@@ -116,6 +117,7 @@ cc_binary(
     name = "protoc_nowkt",
     copts = COPTS,
     linkopts = LINK_OPTS + PROTOC_LINK_OPTS,
+    linkshared = True,
     visibility = [
         "//src/google/protobuf:__pkg__",
     ],
